### PR TITLE
Rafactor translations

### PR DESCRIPTION
**Pull request by 7alip**

### CODE REVIEW

**Review by myniqx — APPROVED** [review]

File: packages/ui/src/components/TranslateLocales/TranslateLocales.tsx [hunk 1]
```diff
@@ -83,259 +61,189 @@ export const TranslateLocales: FC<TranslateLocalesProps> = ({ searchTerm }) => {
...
+  // calculate priority
+  const priorityKeys = useMemo(() => {
+    const updatedPriorityKeys = initialPriorityKeys.map(priorityKey => {
+      const { key } = priorityKey
+      let { priority } = priorityKey
+
+      if (suppressWarning.includes(key)) return priorityKey
+
+      if (!en[key] && !tr[key] && !nl[key]) {
+        priority = PriorityFilter.UNTRANSLATED
```
[the remark this answers is not in the document]
Comment (7alip) [reply packages/ui/src/components/TranslateLocales/TranslateLocales.tsx:73]: Correct! We should probably remove it

File: packages/ui/src/components/TranslateLocales/EntryInput.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const value = targetDict[localeKey]
+
+  const { locked } = useContext(DictContext)
+
+  const [color, setColor] = useState(
+    defaultVal !== value ? 'green.500' : 'gray.300',
+  )
+
+  const handleChange: ChangeEventHandler<HTMLInputElement> = e => {
+    // TODO: Shouldn't we avoid mutating the original object?
```
[the remark this answers is not in the document]
Comment (7alip) [reply packages/ui/src/components/TranslateLocales/EntryInput.tsx:21]: I don't know how useState works without set function but probably your implementation was fine. I was refactoring the code to fix the performance issues and should have restore this changes.

**Comment by myniqx** [comment]
> @myniqx Would you like to fix the issues you notice here?

@7alip done.

**Review by 7alip — COMMENTED** [review]
@myniqx  Thanks, you can merge it

**Pull request merged**